### ISSUE TITLE
parsecfg.writeConfig writes empty file and raises Unknown IOError 

### ISSUE THREAD

**Issue by ftsf** [nim-lang/Nim#5816]
This does not occur in 0.16.0 but occurs in devel

```nim
import parsecfg

var config: Config
try:
  config = loadConfig("test.ini")
except IOError:
  config = newConfig()
config.setSectionKey("General","foo","bar")
config.writeConfig("test.ini")
```

```
Traceback (most recent call last)
cfg.nim(9)               cfg
parsecfg.nim(545)        writeConfig
parsecfg.nim(500)        writeConfig
streams.nim(164)         writeLine
streams.nim(154)         write
streams.nim(130)         writeData
streams.nim(394)         fsWriteData
sysio.nim(65)            checkErr
sysio.nim(57)            raiseEIO
system.nim(2643)         sysFatal
Error: unhandled exception: Unknown IO Error [IOError]
Error: execution of an external program failed: 'cfg '
```
Creates an empty file and errors.
Error only occurs if the Config has content.

Tested on linux, ran same code with stable and the output is correct.

**Comment by cdunn2001** [comment]
I came here to post exactly this. Just update `devel` at `a3e0a7ebb..6bbf0fb64` and this still happens.

**Comment by cdunn2001** [comment]
```nim
539 proc writeConfig*(dict: Config, filename: string) =
540   ## Writes the contents of the table to the specified configuration file.
541   ## Note: Comment statement will be ignored.
542   let file = open(filename, fmWrite)
543   let fileStream = newFileStream(filename)
544   defer: fileStream.close()
545   dict.writeConfig(fileStream)
```
`file` is not used, and `fileStream` is not opened for _write_.

**Comment by cdunn2001** [comment]
Bug introduced here: 3374bcb3736379cbf42663ca9ddbc15053b27f3c

That's a month ago, so we're missing test-coverage on parsecfg too.

And it was a direct push, not part of a pull-request:
```
*   0c694d206 Merge pull request #5564 from nim-lang/fix/4556
|\
| * 34c34cb49 move the object construction logic to a separate file
| * f162214d5 object construction: test cases and manual additions
| * 564c0acae cleaned up the code and implemented proper error messages
| * 02243410f News items for previous commit
| * 6edb07091 fix #4556
* | ab3f22fce minor update for nimsuggest
* | 3d4f68c82 Make nimsuggest quit when pressing Ctrl-D from the command line. (#
* | 3374bcb37 parsecfg: handle streams and strings when saving or loading configu
* | 4ba6f9808 Update 'compiles' documentation to remove not (#5666)
|/
* 88c4d6aab fix another compiler regression
```

I'll submit a PR soon if no-one else says they are working on this...

**Comment by cdunn2001** [comment]
Relevant: #5588
